Source/DafnyCore/AST/Statements/Assignment/AllocateArray.cs

```diff
@@ -9,33 +9,35 @@ namespace Microsoft.Dafny;
 /// Either ElementInit or InitDisplay will be set
 /// </summary>
 public class AllocateArray : TypeRhs, ICloneable<AllocateArray> {
-  public Type EType;
+  public Type ExplicitType;
   public readonly List<Expression> ArrayDimensions;
   public readonly Expression? ElementInit;
   public readonly List<Expression>? InitDisplay;
 
   [SyntaxConstructor]
-  public AllocateArray(IOrigin origin, Type type, List<Expression> arrayDimensions, Expression elementInit)
-    : base(origin) {
+  public AllocateArray(IOrigin origin, Type explicitType, List<Expression> arrayDimensions, Expression elementInit,
```

Comment:
Related to the other comment in this file - if it's true that both `ElementInit` and `InitDisplay` may be null, then the `elementInit` parameter should be nullable.
```suggestion
  public AllocateArray(IOrigin origin, Type explicitType, List<Expression> arrayDimensions, Expression? elementInit,
```